feat: add assignment color codes to Progress API

Serj-N opened the pull request:
### Summary
This PR adds an `assignment_colors` field to the grading policy serializer used by the course progress API.

Serj-N committed:
feat: change list of color codes

openedx-webhooks commented:
Thanks for the pull request, @Serj-N!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

edschema commented:
The list of colors is missing the last one: #3BA03B. The ordered list is below:

#D24242, #7B9645, #5A5AD8, #B0842C, #2E90C2, #D13F88, #36A17D, #AE5AD8, #3BA03B

Serj-N committed:
fix: add missing color

Serj-N commented:
@edschema The missing color has been added.

stan-neichev commented:
@e0d hi, can you please review or assign someone for this PR?

e0d commented on lms/djangoapps/course_home_api/progress/serializers.py:
```diff
@@ -95,8 +95,22 @@ class GradingPolicySerializer(ReadOnlySerializer):
     """
     Serializer for grading policy
     """
+
```
Let's change the name of this and add some comments with a TODO to explain the eventual plan.  So, I'd remove "TEMPORARY."  I'm also not sure what "MOCK" means here.  Can you clarify.

Serj-N replied:
The `MOCK` part is supposed to mean that the color values are hardcoded here and are not meant to be selected/customized by the instructor. I agree that it can cause confusion, so it is better to omit it as well.
I renamed the attribute to `ASSIGNMENT_COLORS`.

UPD: As for a todo, I don't think there is much to add: the backend logic is done, no additional frontend logic is planned, and the mobile solution is already implemented and merged.

Serj-N committed:
style: rename attribute for assignment color list

feanil reviewed (commented):
This seems like a frontend concern, why do we need to provide this via the API?

Serj-N commented:
@feanil Initially, this functionality was indeed designed as a frontend feature, but somewhere along the way, the plans changed, and the implementation started for a full-fledged backend solution. Then the plans changed again, and we ended up with the solution that we've got now.

I believe @cmltaWt0 can provide more details if necessary.

feanil commented:
Yes, I'd like to understand why we need to track the colors here, these seem like values that should live as CSS variables with semantic names but I admin I don't know where these are going to be used.  Is there some UI diagrams for how these values are going to be used?

cmltaWt0 commented:
> Yes, I'd like to understand why we need to track the colors here, these seem like values that should live as CSS variables with semantic names but I admin I don't know where these are going to be used. Is there some UI diagrams for how these values are going to be used?

@feanil this PR doesn't represent the planned implementation at all.
All CSS should be handled by design tokens, this is not implemented yet in Paragon.
@PKulkoRaccoonGang please correct me if I'm wrong.

Backend should work only with Labels. The lasso feature is designed for Instructors to be able to select desired range of color labels. So the final feature design includes the dynamic Label assignment per course (not CSS codes but color Labels), platform-level configuration etc.

Together with @e0d @edschema we decided not to go into implementation for the full feature and do some testing with hardcoded values.

I anticipate that we will want to adjust colors after a feedback and I do not want to go to web, ios, android and make changes in three platforms. So the central place is good as a source of truth. Also it can be easily updated via SiteConfig (need to add this override).

So you concern why do we need an API - because it's designed for a complex Lasso management feature.
Second concern why do we hardcode HEX instead of Labels - because we do not have paragon component in place.
Alternatively we could create big map for each platform and later @edschema can choose from those colors without rebuilding ios and android. But IMHO it's overkill for this simple hypotheses testing.

@edschema Could you comment on this? Obviously I (as a developer) do not like this hardcoding and CSS usage but we do it for testing purposes.

feanil commented:
Thanks for the explaination @cmltaWt0 

@Serj-N I think some of this explaination and links to relevant tickets for this decision and links to tickets for cleaning this up should go into the code as comments next to this code.  I won't be the last person to see this and ask why it's here and we should make sure the answer is available close to where we're adding this since it sounds like it's temporary until the upstream paragon design token variables exist?

Serj-N committed:
docs: add explanation for current implementation

Serj-N commented:
@feanil I added some explanation as comments to the code. As for links to tickets, no ADRs exist for this, there is a [proposal](https://openedx.atlassian.net/wiki/spaces/COMM/pages/5118853122/Proposal+Paragon+Label+Colors+Addition+v2), and all of the decision discussions with @e0d were handled in a google doc.

feanil commented:
FYI, i'm waiting on the above changes before we merge this.

Serj-N commented:
@feanil Hi, I already added a commit with a comment explaining the current implementation, as you requested. And I also added a comment [here](https://github.com/openedx/edx-platform/pull/37438#issuecomment-3585162569) explaining that there isn't much to add in terms of tickets or ADRs, as all of the discussions so far were held in a google doc.

feanil commented:
@Serj-N thanks, can you add a link to the google doc here as well so that this ticket can serve as the single point of context.

edschema commented:
Hi! Here is the original [scoping doc](https://docs.google.com/document/d/1T7e8dDuCkzw8oSg6U5n5QiZ_6mO7lmpf0o-JFg7STvM/edit?tab=t.q666w042frfi#heading=h.jvvxby91630b) that Raccoon Gang had written up

Adding notes here to give visibility to the decision to hard code the values.
Initially the color coding was imagined as:
- Adding label colors to Paragon to ingest for assignment types
- The set lasso on the Studio Grading page was meant to allow users to select colors in a set since the label colors are ordinal. This was descoped early on
- After this exploration, RG did a secondary exploration of how we could a set of default/most common assignment types with a set of the label colors and use other colors from the label colors for any additional assignment types. This was descoped because it was a much higher lift than anticipated

Descoping the above means instead of ingesting colors, we will instead hard-code the values that are based on the [Paragon proposal](https://openedx.atlassian.net/wiki/spaces/COMM/pages/5118853122/Proposal+Paragon+Label+Colors+Addition+v2). In the future we hope that we will use ingested colors instead of hard coded values but for now this is where we landed

e0d commented:
Thanks @edschema ! @feanil would you like to merge?
